### PR TITLE
Firefox only partially supports `background-clip: text`

### LINKED ISSUE #23943: css.properties.background-clip - though browsers support `text` value, the spec still says it's experimental
### What type of issue is this?

Other

### What information was incorrect, unhelpful, or incomplete?

- related to MDN issue https://github.com/mdn/content/issues/26445#issuecomment-2247413333
- refer discussion https://github.com/w3c/csswg-drafts/issues/10595

The `text` value for `background-clip` has been introduced in [CSS Backgrounds Module Level 4](https://drafts.csswg.org/css-backgrounds-4/#valdef-background-clip-text) and the specification is not ready for implementation yet. So the `text` value hasn't been ironed out yet.

### What browsers does this problem apply to, if applicable?

Chromium (Chrome, Edge 79+, Opera, Samsung Internet), Firefox, Safari

### What did you expect to see?

Mark the `text` value of the `background-clip` property as `non-standard` or at least `experimental`.

### Did you test this? If so, how?

Testing the issue https://github.com/mdn/content/issues/26445 shows that the value hasn't been ironed out yet.

### Can you link to any release notes, bugs, pull requests, or MDN pages related to this?

https://github.com/mdn/content/issues/26445

### Do you have anything more you want to share?

_No response_

### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/background-clip

### MDN metadata

<details>
<summary>MDN page report details</summary>

* Query: `css.properties.background-clip`
* Report started: 2024-07-24T09:51:26.390Z

</details>

### PR DESCRIPTION
**Pull request by SelenIT**
#### Summary

Mark background-clip:text as experimental (since it's only in the CSS Backgrounds 4 editor's draft) and mark Firefox's implementation as experimental because of [many open issues](https://bugzilla.mozilla.org/show_bug.cgi?id=1656784) related to it.

Add background-clip:border from the same spec for consistency.

#### Test results and supporting details

None

#### Related issues

Fixes #23943

### CODE REVIEW

**Comment by SelenIT** [comment]
The auto tests script states that "Experimental should be set to false as the feature is supported in multiple browser engines" and marks experimental status for `background-clip:text` as "Error", but I believe this particular case is a valid exception from this rule, as the "should" keyword is supposed to allow.

The `text` value for `background-clip` property was unilaterally added by WebKit as a non-standard extension and became widely supported only because WebKit and its derivative Blink engines became so popular. The CSSWG has [opposed](https://github.com/w3c/csswg-drafts/issues/9907#issuecomment-1930785584) standardizing this feature for years, it was never properly specified. Firefox's implementation still lacks some important use cases like multi-layer backgrounds combining `text` with other `background-clip` values for other layers. Since this value is in the spec draft now, it can't be marked as non-standard anymore, but marking it non-experimental only because of popularity of the clones of the engine that introduced this _experimental_ feature doesn't feel right.

I would suggest keeping its experimental status until the CSS Backgrounds Level 4 spec matures and/or Firefox's implementation issues are fixed, if it's possible.

File: css/properties/background-clip.json [hunk 1]
```diff
@@ -101,6 +104,42 @@
             "deprecated": false
           }
         },
+        "border": {
+          "__compat": {
+            "spec_url": "https://drafts.csswg.org/css-backgrounds-4/#valdef-background-clip-border",
```
Comment (yisibl) [inline css/properties/background-clip.json:109]: https://drafts.csswg.org/css-backgrounds-4/#valdef-background-clip-border-area

File: css/properties/background-clip.json [hunk 2]
```diff
@@ -101,6 +104,42 @@
             "deprecated": false
           }
         },
+        "border": {
```
Comment (yisibl) [inline css/properties/background-clip.json:107]: Note that it has been renamed to `border-area` in the specification.

File: css/properties/background-clip.json [hunk 3]
```diff
@@ -302,7 +305,9 @@
                 }
               ],
               "firefox": {
-                "version_added": "49"
+                "version_added": "49",
+                "partial_implementation": true,
+                "notes": "Does not work with <code>background-attachment:fixed</code>, in multi-layer backgrounds with other values, and in some other special cases: see <a href='https://bugzil.la/1656784'>bug 1656784</a>"
```
Comment (caugner) [inline css/properties/background-clip.json:310]:
```suggestion
                "notes": "Does not work with <code>background-attachment: fixed</code>, in multi-layer backgrounds with other values, and in some other special cases: see <a href='https://bugzil.la/1656784'>bug 1656784</a>"
```

Comment (caugner) [reply css/properties/background-clip.json:310]: Can you add the specific bugs to the mentioned issues, to make it easier to check later if these have been resolved?

> `<code>background-attachment: fixed</code>`


https://bugzilla.mozilla.org/show_bug.cgi?id=1313757

> `in multi-layer backgrounds with other values`

https://bugzilla.mozilla.org/show_bug.cgi?id=1908386 ?

> `and in some other special cases`

Here you can keep the meta bug.

**Review by caugner — CHANGES_REQUESTED** [review]
Note that `background-clip: border-radius` was meanwhile added via https://github.com/mdn/browser-compat-data/pull/24333.

The IE note fixes make sense.

As for the Firefox comment, let's add specific bug links to the mentioned issues.

File: css/properties/background-clip.json (hunk 3 above)
Comment (SelenIT) [reply css/properties/background-clip.json]: Done, link text formatted according to CI rules requirements

**Review by caugner — APPROVED** [review]

**Review by caugner — APPROVED** [review]

**Pull request merged**